### DISCUSSION TITLE
Consider moving Psalm\Tests\ from autoload-dev to autoload in composer.json

### DISCUSSION THREAD

**Discussion by orklah** [vimeo/psalm#4887] [Ideas]
Hi,

I'm writing a plugin for Psalm and I would like to use the same test system as Psalm to launch my tests. (The alternative being launching a full psalm instance by command line).

For that, I tried to use some Fake classes created especially for test usage but they're in the namespace Psalm\Tests\ who is only available in autoload-dev.

I tried copy/pasting those three classes:
FakeFileProvider.php
FakeParserCacheProvider.php
TestConfig.php

And I could validate my tests correctly. If this suggestion is not accepted, would you allow me to copy/paste those classes in my plugin as-is?

**Comment by weirdan** [comment]
> The alternative being launching a full psalm instance by command line

Shameless plug: If you decide to use this approach instead, there's [Codeception module](https://github.com/weirdan/codeception-psalm-module) that makes it almost pleasant experience.
